pandas/_libs/tslibs/offsets.pyx

```diff
@@ -4615,21 +4616,22 @@ cpdef to_offset(freq, bint is_period=False):
 
             tups = zip(split[0::4], split[1::4], split[2::4])
             for n, (sep, stride, name) in enumerate(tups):
-                if is_period is False and name == "M":
+                if is_period is False and name in c_OFFSET_DEPR_FREQSTR:
                     warnings.warn(
-                        "\'M\' will be deprecated, please use \'ME\' "
-                        "for \'month end\'",
+                        f"\'{name}\' will be deprecated, please use "
+                        f"\'{c_OFFSET_DEPR_FREQSTR.get(name)}\' for Offsets.",
```

Comment:
maybe we can just say `f"\'{c_OFFSET_DEPR_FREQSTR.get(name)}\' instead."`?

Reply:
Thank you, sounds good to me. I corrected the message for UserWarning.